### PR TITLE
use hasql-pool with connection lifetime option

### PR DESCRIPTION
**Pull request by robx**
This uses the new hasql-pool-0.9 to add `db-pool-max-lifetime`.

### CODE REVIEW

File: src/PostgREST/CLI.hs [hunk 1]
```diff
@@ -151,6 +146,9 @@ exampleConfigFile =
       |## Time in seconds to wait to acquire a slot from the connection pool
       |# db-pool-acquisition-timeout = 10
       |
+      |## Time in seconds after which to recycle pool connections
+      |# db-pool-connection-lifetime = 3600
```
[the remark this answers is not in the document]
Comment (steve-chavez) [reply src/PostgREST/CLI.hs:150]: Ah, just saw [pgbouncer's server_lifetime](https://www.pgbouncer.org/config.html#server_lifetime)(btw, this has the same behavior as `db-pool-connection-lifetime` right?) default is 3600, perhaps it's related to this value?

File: src/PostgREST/Config.hs [hunk 2]
```diff
@@ -71,6 +71,7 @@ data AppConfig = AppConfig
   , configDbPlanEnabled            :: Bool
   , configDbPoolSize               :: Int
   , configDbPoolAcquisitionTimeout :: Maybe Int
+  , configDbPoolConnectionLifetime :: Maybe Int
```
[the remark this answers is not in the document]
Comment (robx) [reply src/PostgREST/Config.hs:74]: I think `Int` vs `Maybe Int` and setting a default are separate issues.

Currently:
- type is `Maybe Int`, which allows for the timeout to be disabled
- default value is `Nothing`, which means by default it is disabled

Enable this by default, but allow disabling:
- type is `Maybe Int`
- default value is (e.g.) `Just 3600`

Require this to be enabled (and necessarily provide a default)
- type is `Int`
- default value is (e.g.) `3600`

Happy to do whichever you like

[the remark this answers is not in the document]
Comment (steve-chavez) [reply src/PostgREST/Config.hs:74]: Right, got it.

> (You might want to consider the acquisition timeout also while we're at it -- would you like it to be enabled by default, and would you like to allow disabling it?)

Yes, thanks for the reminder. Let's put it at 10 seconds by default.

File: src/PostgREST/CLI.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (steve-chavez) [reply src/PostgREST/CLI.hs:150]: Let's leave it at 30 mins. Btw, I've seen a [discussion around HikariCP maxLifetime on the pg mailing list](https://www.postgresql.org/message-id/flat/CA%2Bmi_8bnvpxHZtb6EgHSHY-xn29W8VJMzjPU3fiCOv1bfjrNuA%40mail.gmail.com), pg hackers are very positive about having a default lifetime.

[the remark this answers is not in the document]
Comment (robx) [reply src/PostgREST/CLI.hs]: I've changed this to 30m now.

File: src/PostgREST/Config.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (robx) [reply src/PostgREST/Config.hs]: Done (setting acq timeout to 10s by default, max lifetime 30m, both non-optional -- turns out that setting a non-Nothing default implies making the setting required)

**Review by steve-chavez — APPROVED** [review]
Great work! :fire:

**Pull request merged**